tools/clang/test/HLSLFileCheck/dxil/intrinsics/launch_types_mesh.ll

```diff
@@ -0,0 +1,307 @@
+; RUN: %dxilver 1.8 | %dxv %s
+; This test's content is from the compilation output of 
+; \tools\clang\test\HLSLFileCheck\d3dreflect\rdat_mintarget\sm69_mesh_node_deriv.hlsl
+; compiling with -T lib_6_9. 
+; The purpose of the test is to detect validation errors when using 
+; derivative operations on node shaders with the mesh launch type.
+; No diagnostics are expected, because derivatives are expected to be allowed in mesh nodes.
+
+; Ensure that categories of deriv ops are allowed for mesh node shaders.
+; Ensure that the OptFeatureInfo_UsesDerivatives flag is set as well.
```

Comment:
What is this test for exactly?  You can test this from HLSL compiling through to DXIL and check the DXIL for expected properties, right?  It's not a validator test, because there's nothing expected from the validator here.  An end-to-end HLSL to DXIL test will verify that the validator passes as well.  The hlsl file referenced does this already.

It may be fine to keep this, but I'm not sure it's necessary, and it would need re-generating anyway:

We expect the `DerivativesInMeshAndAmpShaders` flag to be set, which should expect to see `"Derivatives in mesh and amplification shaders"` in a comment in the DXIL disassembly after `Note: shader requires additional functionality:`.

You say here "Ensure that the OptFeatureInfo_UsesDerivatives flag is set as well.", but we do not expect that flag to be set in module flags (only in RDAT).  That flag is filtered out of the module flags (it's a flag for functions used in RDAT for linkage validation, but isn't present in the global module flags).  This test doesn't check that, and it isn't present in the module.

The module flags are later in the metadata: `8590458896` or hex: `0x200080010`, which breaks down into the DxilShaderFlags: `ResMayNotAlias` (default when not passing -res-may-alias), `WaveOps` (due to Quad op), and `EnableRawAndStructuredBuffers` (using a raw buffer).

Reply:
Yes, the validator does run on sm69_mesh_node_deriv, so I shouldn't need this test at all. I'll get rid of it.